### PR TITLE
[WIP] Changes to bring builder into automate

### PR DESCRIPTION
**Pull request by jaym**
## Add SSL configuration for depot postgres client
To bring this into automate, depot needs to:
- use SSL to connect to postgres
- use username/password auth (for automate-cluster)
- use TLS authentication for standalone automate

This change doesn't address builder-jobsrv and builder-graph. For some reason, they create their database pools in a different way. I'm not sure if I can change it to be more like builder-api

## builder-web needs to be able to run at /bldr
- We need to be able to prepend /bldr to paths

For some reason, base href doesn't work. Also, relative paths don't seem to work inside the custom elements.

### CODE REVIEW

File: components/builder-db/src/config.rs [hunk 1]
```diff
@@ -49,7 +53,11 @@ impl Default for DataStoreCfg {
                        connection_retry_ms:    300,
                        connection_timeout_sec: 3600,
                        connection_test:        false,
-                       pool_size:              (num_cpus::get() * 2) as u32, }
+                       pool_size:              (num_cpus::get() * 2) as u32,
+                       ssl_mode:               None,
```
Comment (chefsalim) [inline components/builder-db/src/config.rs]: Could you also please add a test for these configs here - https://github.com/habitat-sh/builder/blob/master/components/builder-api/src/config.rs#L260

File: components/builder-db/src/config.rs [hunk 2]
```diff
@@ -65,6 +73,28 @@ impl fmt::Display for DataStoreCfg {
             None => connect,
         };
         connect = format!("{}@{}:{}/{}", connect, self.host, self.port, self.database);
+        let mut opts = Vec::new();
+
+        if let Some(ref m) = self.ssl_mode {
+            opts.push(format!("sslmode={}", m));
```
Comment (chefsalim) [inline components/builder-db/src/config.rs]: Do we also need to propagate this to IntoConnectParams below?

Comment (jaym) [reply components/builder-db/src/config.rs]: From what I can tell, that isn't possible as you have to create special types that can read/write on ssl connections. I'm not sure why the 2 pools use different types.

Comment (jaym) [reply components/builder-db/src/config.rs]: specifically, the problematic bits look like they are here:
https://github.com/habitat-sh/builder/blob/master/components/builder-db/src/pool.rs#L45-L47

the ssl-ness is in a different parameter

Comment (chefsalim) [reply components/builder-db/src/config.rs]: Yeah there are 2 pools .. one for Diesel queries, and one for non-Diesel (legacy).
So we should probably fix that line above with the way we have the Diesel pool creation at https://github.com/habitat-sh/builder/blob/master/components/builder-db/src/diesel_pool.rs#L37

**Comment by chefsalim** [comment]
@jaym The suggested changes above should be straightforward, please let us know if you need assistance to get this updated/merged.

**Comment by jaym** [comment]
we're closing this for now while we work on getting a list of all the changes that would be needed

**Pull request closed without merging**